app/docs/0.11.x/plugin-development/entities-cache.md

```diff
@@ -15,195 +15,275 @@ chapter: 7
-that data, it will find the data in the SHM, deserialize it (and store it in its own
-Lua memory cache) and then return it.
+When data is fetched from the database, it will be stored in both caches. Now
+if the same worker process requests the data again, it will retrieve the
+previously-deserialized data from the Lua memory cache. If a different
+worker within the same nginx node requests that data, it will find the data
+in the SHM, deserialize it (and store it in its own Lua memory cache) and
+then return it.
 
 This module exposes the following functions:
 
-| Function name                      | Description
-|------------------------------------|---------------------------
-| `ok, err = cache.set(key, value, ttl)`  | Stores a Lua object into the in-memory cache with the specified key (the optional ttl is in seconds). The value can be any Lua type, including tables. Returns `true` or `false`, and `err` if the operation fails.
-| `value = cache.get(key)`           | Retrieves the Lua object stored in a specific key.
-| `cache.delete(key)`                | Deletes the cache object stored at the specified key.
-| `ok, err = cache.sh_add(key, value, ttl) | Adds a new value to the SHM cache, optional ttl in seconds (if `nil` it never expires)
-| `ok, err = cache.sh_set(key, value, ttl) | Sets a new value under the specified key in the SHM, optional ttl in seconds (if `nil` it never expires)
-| `value = cache.sh_get(key)         | returns the value stored in the SHM under the key, or `nil` if not found
-| `cache.sh_delete(key)              | deletes a value from the SHM
-| `newvalue, err = cache.sh_incr(key, amount)` | Increments a number stored in SHM under the specified key, by an amount of units specified. The number needs to be already present in the cache or an error will be returned. If successful, it returns the newly-incremented value, otherwise an error.
-| `value, ... = cache.get_or_set(key, ttl, function, ...)` | This is a utility method that retrieves an object with the specified key, but if the object is `nil` then the passed function will be executed instead, whose return value will be used to store the object at the specified key. This effectively makes sure that the object is only loaded from the datastore one time, since every other invocation will load the object from the in-memory cache.
+Function name                                 | Description
+----------------------------------------------|---------------------------
+`value, err = cache:get(key, opts?, cb, ...)` | Retrieves the value from the closest cache level. If no cache (L1 nor L2) has the value, invokes `cb` in protected mode. `cb` must return one (and only one) value that will be cached. It *can* throw errors, as those will be caught and properly logged by Kong, at the `ngx.ERR` level. This function **does** cache negative results (`nil`). As such, one must rely on its second argument `err` when checking for errors.
```

Comment:
I'm afraid this format is really not appropriate to document `opts` :( So I left it out... It's documented in the upcoming release of lua-resty-mlcache for now (internal or advanced documentation maybe). I think ideally once we have some more time, we can improve this document to instead point to a more complete reference of the available API...